### PR TITLE
[SELS-TASK] Created Lesson CRUD with Gates, and Admin User management API

### PR DESCRIPTION
**Pull request by johnpaul-sun**
`[Asana Link/s]`
- https://app.asana.com/0/1202335275873837/1202335275873859/f

`[Commands]`
From the root folder
- cd back-end
- php artisan serve
- php artisan migrate:fresh --seed

`[Pre-conditions]`
Open this link -> http://127.0.0.1:8000/
Go to Postman or Insomnia to test the API

`[Expected Output]`

### For User  
| Route | Method     | Description                |
| :-------- | :------- | :------------------------- |
| `/api/v1/users` | `POST` | Register new User in the Database|
| `/api/v1/users/login` | `POST` | Login user and generate Token for Authentication | 

### For User 
| Route | Method     | Description                |
| :-------- | :------- | :------------------------- |
| `/api/v1/users` | `GET` | Get all registered user in the Database excluding Admin|
| `/api/v1/users/{user_id}` | `GET` | Get specific user based on id from the database | 
| `/api/v1/users/{user_id}` | `PUT` | Update specific user based on id from the database | 
| `/api/v1/users/logout` | `POST` | Logout User and destroy the Authentication Token | 

### For Lesson 
| Route | Method     | Description                |
| :-------- | :------- | :------------------------- |
| `/api/v1/lessons` | `POST` | Create new lesson|
| `/api/v1/lessons` | `GET` | Get all lessons in the Database|
| `/api/v1/lessons/{lesson_id}` | `GET` | Get specific lesson based on id from the database | 
| `/api/v1/lessons/{lesson_id}` | `PUT` | Update specific lesson based on id from the database | 
| `/api/v1/lessons/{lesson_id}` | `DELETE` | Delete specific lesson based on id from the database | 

### For Admin 
| Route | Method     | Description                |
| :-------- | :------- | :------------------------- |
| `/api/v1/admin` | `GET` | Get all registered user in the Database excluding Admin|
| `/api/v1/admin/{user_id}` | `GET` | Get specific user based on id from the database | 
| `/api/v1/admin/{user_id}` | `PUT` | Update specific user based on id from the database | 
| `/api/v1/admin/{user_id}` | `DELETE` | Delete specific user based on id from the database |  

 
`[Notes]`

- Ensure to add this in the header `Accept application/json` in all methods to avoid the 405 methods not allowed error.
- The CRUD for Questions and Choices is currently in progress, I pushed this PR because this is only my task for today.

`[Screenshots]`

image

### CODE REVIEW

File: back-end/app/Http/Controllers/Admin/AdminController.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+        User::findOrFail($user_id)->update([
+            'first_name' => $user['first_name'],
+            'last_name' => $user['last_name'],
+            'email' => $user['email'],
+            'password' => bcrypt($user['password']),
+        ]);
+
+        return response()->json(['message' => 'Account Updated Successfully'], 201);
+    }
```
Comment (chevyB) [inline back-end/app/Http/Controllers/Admin/AdminController.php]: I don't think that we need this one. In real world, it is not practical that admin can change user's info.
But in change password, it is possible but it should generate random password.
Else, we can remove this feature.

Comment (chevyB) [inline back-end/app/Http/Controllers/Admin/AdminController.php:57]: I don't think that we need this one. In real world, it is not practical that admin can change user's info.
But in change password, it is possible but it should generate random password.
Else, we can remove this feature.

File: back-end/app/Http/Controllers/Lesson/LessonsController.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        $data = $request->validate([
+            'title' => 'required|string|max:191',
+            'description' => 'required|string|max:191',
+        ]);
+
+        Lesson::findOrFail($lesson_id)->update([
+            'title' => $data['title'],
+            'description' => $data['description'],
+        ]);
```
Comment (chevyB) [inline back-end/app/Http/Controllers/Lesson/LessonsController.php]: This is for update. While your update method has create logic. Swap it nalang.

File: back-end/app/Http/Controllers/Admin/AdminController.php (hunk 1 above)
Comment (johnpaul-sun) [reply back-end/app/Http/Controllers/Admin/AdminController.php]: Noted Sir Chev, I'll remove this method and leave the read and delete users only.

File: back-end/app/Http/Controllers/Lesson/LessonsController.php (hunk 2 above)
Comment (johnpaul-sun) [reply back-end/app/Http/Controllers/Lesson/LessonsController.php]: Oh, I'm sorry, it got switched with the update method. I'll fix all of the comments and re-push.

**Review by chevyB — APPROVED** [review]

**Pull request merged**